packages/js-drive/lib/abci/handlers/finalizeBlock/commitFactory.js

```diff
@@ -66,6 +70,29 @@ function commitFactory(
       }
     }
 
+    // Send withdrawal transactions to Core
+    const unsignedWithdrawalTransactionsMap = blockExecutionContext.getWithdrawalTransactionsMap();
+
+    const { thresholdVoteExtensions } = lastCommitInfo;
+
+    for (const { extension, signature } of (thresholdVoteExtensions || [])) {
+      const withdrawalTransactionHash = extension.toString('hex');
+
+      const unsignedWithdrawalTransactionBytes = unsignedWithdrawalTransactionsMap[
+        withdrawalTransactionHash
+      ];
+
+      if (unsignedWithdrawalTransactionBytes) {
```

Comment:
I would probably throw an error if we get something extra. And vise-versa (didn't get signature for what we requested). Better to discuss with @shotonoff

Reply:
Discussed, should not happen nor less nor extra data.